webapp/src/components/testing/TestSteps.vue

```diff
@@ -177,9 +205,60 @@ const visitedSamples = computed<number>(() => {
     }
   }
 });
+
+const confusionMatrix = computed<{labels : Map<number, string>, matrix : number[][]} | undefined>(() => {
+  if (tested.value === undefined) return undefined;
+  const labels = new Set<number>();
+  const mapLabels = new Map<number, string>();
+  
+  // get all the labels
+  switch (props.task.trainingInformation.dataType) {
+    case "image":
+      (tested.value as Tested["image"]).forEach(({ output }) => {
+        labels.add(output.label);
+        labels.add(output.predicted);
+        mapLabels.set(output.label, output.truth);
+      });
+      break;
+    case "text":
+      return undefined;
+    case "tabular":
+      (tested.value as Tested["tabular"]).results.forEach(({ output }) => {
+        labels.add(output.label);
+        labels.add(output.predicted);
+        mapLabels.set(output.label, output.truth);
+      });
+      break;
```

Comment:
no need to compute for tabular if we don't support it in the end

Reply:
I added the tabular matrix now, but the big chunk of code was to have a complete list of label, which is not used anymore, because there is a better way to do so